boring/src/ssl/mod.rs

```diff
@@ -2748,6 +2764,20 @@ impl SslRef {
             .expect("invalid default server curves list");
     }
 
+    /// Returns the [`SslCurve`] used for this `SslRef`.
+    ///
+    /// This corresponds to [`SSL_get_curve_id`]
+    ///
+    /// [`SSL_get_curve_id`]: https://commondatastorage.googleapis.com/chromium-boringssl-docs/ssl.h.html#SSL_get_curve_id
+    pub fn curve(&self) -> Option<SslCurve> {
+        let curve_id = unsafe { ffi::SSL_get_curve_id(self.as_ptr()) };
+        if curve_id == 0 {
+            return None;
+        }
+        let curve_id: c_int = curve_id.try_into().ok()?;
```

Comment:
Clippy complains about this, is `.into()` doable on all archs or should we silence the lint there?

Reply:
I imagine it is. Updated